### PR TITLE
Add XLA recipe

### PR DESCRIPTION
**Pull request by mofeing**

### CODE REVIEW

File: X/XLA/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/openxla/xla.git", "3e0222a9a4f11b5f47ed6c3bcbbdaeee81e2bc68")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+apk update
+apk add bazel --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing
+apk add py-numpy
```
Comment (giordano) [inline X/XLA/build_tarballs.jl:17]: `py3-numpy`: https://pkgs.alpinelinux.org/package/v3.15/community/x86_64/py3-numpy?
```suggestion
apk add py3-numpy
```

File: X/XLA/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Platform("aarch64", "linux"),
+    Platform("x86_64", "macos"),
+    Platform("aarch64", "macos"),
+]
+
+# The products that we will ensure are always built
+products = Product[]
+
+# Dependencies that must be installed before this package can be built
+dependencies = []
```
Comment (giordano) [inline X/XLA/build_tarballs.jl:41]:
```suggestion
dependencies = Dependency[]
```

**Comment by mofeing** [comment]
Oh no, "no space left on device" error 🥲

**Comment by giordano** [comment]
Uhm, the disk of the build machine is good, full only at 74%, so I think the no space left on device error may refer to the temp memory, which is much harder to control.  I have no idea how to increase that 🥲

**Comment by mofeing** [comment]
No luck, but I've found out that intermediate files can be created in another directory where there is more space with the [--output_base](https://docs.bazel.build/versions/master/command-line-reference.html#flag--output_base) flag. Is there any other place I can point to?

**Comment by giordano** [comment]
```
[11:16:36]  ---> df -hT
[11:16:36] Filesystem           Type            Size      Used Available Use% Mounted on
[11:16:36] overlay              overlay         1.0G      1.0G         0 100% /
[11:16:36] overlay              overlay         1.0G      1.0G         0 100% /opt/aarch64-apple-darwin20
[11:16:36] overlay              overlay         1.0G      1.0G         0 100% /opt/x86_64-linux-musl
[11:16:36] udev                 devtmpfs      251.8G         0    251.8G   0% /dev/null
[11:16:36] udev                 devtmpfs      251.8G         0    251.8G   0% /dev/tty
[11:16:36] udev                 devtmpfs      251.8G         0    251.8G   0% /dev/urandom
[11:16:36] tank/root            zfs           228.6G     25.7G    202.9G  11% /etc/resolv.conf
[11:16:36] /dev/nvme0n1p1       ext4            3.4T      2.4T    871.8G  74% /root/.ccache
[11:16:36] tank/root            zfs           228.6G     25.7G    202.9G  11% /opt/toolchains
[11:16:36] tank/root            zfs           228.6G     25.7G    202.9G  11% /opt/bin
[11:16:36] /dev/nvme0n1p1       ext4            3.4T      2.4T    871.8G  74% /meta
[11:16:36] /dev/nvme0n1p1       ext4            3.4T      2.4T    871.8G  74% /workspace
```
`/` is full, but that's an overlay file system, no idea how to change its size.  @staticfloat ideas?

**Comment by mofeing** [comment]
I believe that its the goal of the `--output_base` flag.

**Pull request closed without merging**